### PR TITLE
Add kube-cluster env for tsh

### LINKED ISSUE #7641: tsh env variable for kube-cluster
### What
Ability to set kube-cluster as a tsh env.

### How
Automatically log into and pin the specified kube-cluster once a user executes `tsh login`

### Why
Customer ask for shops that have contractors/reps that work exclusively in a single kube cluster.

### PR DESCRIPTION
**Pull request by Joerger**
I also refactored the tsh env tests to reuse common logic and converted them into closure-driven tests which I find to be much more readable than long tables.

Closes https://github.com/gravitational/teleport/issues/7641

### CODE REVIEW

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -2191,14 +2188,28 @@ func onEnvironment(cf *CLIConf) error {
 	case cf.unsetEnvironment:
 		fmt.Printf("unset %v\n", proxyEnvVar)
 		fmt.Printf("unset %v\n", clusterEnvVar)
+		fmt.Printf("unset %v\n", kubeClusterEnvVar)
 	case !cf.unsetEnvironment:
 		fmt.Printf("export %v=%v\n", proxyEnvVar, profile.ProxyURL.Host)
```
[the remark this answers is not in the document]
Comment (russjones) [reply tool/tsh/tsh.go:2193]: @timothyb89 Does Windows have the equivalent of environment variables and commands like `unset`?

[the remark this answers is not in the document]
Comment (timothyb89) [reply tool/tsh/tsh.go:2193]: I believe the syntax for Windows varies for cmd.exe vs PowerShell:

cmd.exe:
```
set FOO=bar
set FOO=
```


PowerShell:
```
$Env:FOO = "bar"
Remove-Item Env:\FOO
```

[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tsh/tsh.go:2193]: @timothyb89 you should be able to check for the `PSModulePath` environment variable to see if you're in PowerShell.

**Review by andrejtokarcik — COMMENTED** [review]
Do I understand correctly this PR overrides https://github.com/gravitational/teleport/pull/7840 in the sense that Teleport should _not_ take the `$KUBECONFIG` env var into account and use `$TELEPORT_KUBE_CLUSTER` instead? If so, shouldn't `kubeconfig.PathFromEnv` be removed?

**Comment by Joerger** [comment]
> Do I understand correctly this PR overrides #7840 in the sense that Teleport should _not_ take the `$KUBECONFIG` env var into account and use `$TELEPORT_KUBE_CLUSTER` instead? If so, shouldn't `kubeconfig.PathFromEnv` be removed?

This PR doesn't override the other one. `$TELEPORT_KUBE_CLUSTER` is just used to set `CLIConf.KubernetesCluster` rather than providing it through `tsh login --kube-cluster=...`. So actually the two PRs can be used together by setting `$TELEPORT_KUBE_CLUSTER=kube1` and `$KUBECONFIG=~/.tsh/.../kubeconfig-kube1`.

**Review by tcsc — APPROVED** [review]

**Review by andrejtokarcik — APPROVED** [review]

File: tool/tsh/tsh.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tsh/tsh.go:2251]: @Joerger let's just make a separate issue to track this.

File: tool/tsh/tsh.go [hunk 2]
```diff
@@ -2303,3 +2314,17 @@ func readTeleportHome(cf *CLIConf, fn envGetter) {
 		cf.HomePath = path.Clean(homeDir)
 	}
 }
+
+// readKubernetesClusterFlag figures out the kube cluster the user is attempting to select.
+// Command line specification always has priority, after that TELEPORT_KUBE_CLUSTER.
+func readKubernetesClusterFlag(cf *CLIConf, fn envGetter) {
```
Comment (zmb3) [inline tool/tsh/tsh.go:2320]: Might want to rename this since it does more than than read a flag.

File: tool/tsh/tsh_test.go [hunk 3]
```diff
@@ -592,68 +592,118 @@ func TestFormatConnectCommand(t *testing.T) {
...
+		t.Run("TELEPORT_KUBE_CLUSTER set", testEnvFlag(testCase{
+			inCLIConf: CLIConf{},
+			envMap: map[string]string{
+				kubeClusterEnvVar: "a.example.com",
+			},
+			outCLIConf: CLIConf{
+				KubernetesCluster: "a.example.com",
+			},
+		}))
+		t.Run("TELEPORT_KUBE_CLUSTER and CLI flag is set, prefer CLI", testEnvFlag(testCase{
```
Comment (zmb3) [inline tool/tsh/tsh_test.go:678]: Add a test for only the CLI flag set.

**Review by zmb3 — CHANGES_REQUESTED** [review]
Would be good to see one more test to cover all 4 cases (nothing set, flag set, env var set, both flag and env var set).

File: tool/tsh/tsh.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Joerger) [reply tool/tsh/tsh.go:2252]: @zmb3  https://github.com/gravitational/teleport/issues/8047

**Review by zmb3 — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Pull request merged**